app/models/chef.rb

```diff
@@ -1,4 +1,18 @@
 class Chef < ApplicationRecord
    validates_presence_of :name
    has_many :dishes
-end
\ No newline at end of file
+
+
+   def ingredient_list
+   ingredients = []
+      self.dishes.each do |dish|
+         ingredients << dish.ingredients.pluck(:name)
+      end
+      ingredients.flatten.uniq
+   end
```

Comment:
What you're currently doing is something called an N+1 query which is something that should be avoided. We make 1 initial query to the database, `self.dishes`, and then for every dish we loop through and make a second query to get the ingredients.

Using pure AR, this could be refactored to something like this:

```ruby
def ingredient_list
  Chef.joins(dishes: :ingredients).where(dishes: { chef_id: self.id }).select("ingredients.name").distinct.pluck(:name)
end
```

Now while this works, this is a little verbose. We could shorten it to something like this:

```ruby
has_many :ingredients, through: :dishes

def ingredient_list
  self.ingredients.distinct.pluck(:name)
end
```

With this new association we can go just call on `ingredients` and let ruby handle the `joins` and the `where` behind the scenes. Let me know if this doesn't make sense and we can go over it some more together.